app/models/section.rb

```diff
@@ -1,8 +1,8 @@
 class Section < ActiveRecord::Base
-  belongs_to :course
-  validates  :name, presence: true, uniqueness: { scope: :course_id }
+  has_and_belongs_to_many :instructor
+  validates  :shortname, presence: true, uniqueness: { scope: :listing_id }
   validates  :crn, presence: true, uniqueness: true
```

Comment:
this validation will break multiple semesters. CRNs are unique in the scope of a session only.